ext/mbstring/php_mbregex.c

```diff
@@ -1051,28 +1053,25 @@ static void _php_mb_regex_ereg_replace_exec(INTERNAL_FUNCTION_PARAMETERS, OnigOp
 		}
 
 		if (!php_mb_check_encoding(string, string_len, enc)) {
-			RETURN_NULL();
+			zend_argument_value_error(3, "must be a valid string in '%s'", php_mb_regex_get_mbctype());
```

Comment:
I don't think we should throw on incorrectly encoded string, as this is likely user-provided data.